src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -256,8 +256,11 @@ public void TestColumnDefinitionNegativeCases()
         /// <summary>
         /// Validates serialization and deserilization of Dictionary containing DatabaseTable
         /// this is how we serialize and deserialize metadataprovider.EntityToDatabaseObject dict
+        /// [Test Ignored] See https://github.com/dotnet/runtime/issues/41920#issuecomment-688497802
```

Comment:
Yes, removed because this breaks in .net8. 

The custom DatabaseObjectConverter experiences deserialization issue, I believe, due to 
- Breaking Change: https://learn.microsoft.com/en-us/dotnet/core/compatibility/serialization/8.0/metadata-resolving

Here is a diff of the deserialized json text that the test creates and feeds to the converter:
![image](https://github.com/Azure/data-api-builder/assets/6414189/b5abe274-ae3f-4db7-954c-47c2312b5d9c)

![image](https://github.com/Azure/data-api-builder/assets/6414189/7d39263b-1274-4a79-81e2-0342f5809ead)

Our tests run for both .net8 and .net6 target frameworks. However, locally in visual studio , even though the TestExplorer shows tests for both target frameworks, this particular test fails for .net6 when I have dotnet8 installed. I don't know why that is because i'd expect .net6 tests to use .net6 runtime/sdk.

Here is sample pipeline logs that show tests running for both target frameworks

> Test run for D:\a\1\s\src\out\tests\net8.0\Azure.DataApiBuilder.Service.Tests.dll (.NETCoreApp,Version=v8.0)
> Passed!  - Failed:     0, Passed:  1162, Skipped:     2, Total:  1164, Duration: 8 m 59 s - Azure.DataApiBuilder.Service.Tests.dll (net8.0)

> Test run for D:\a\1\s\src\out\tests\net6.0\Azure.DataApiBuilder.Service.Tests.dll (.NETCoreApp,Version=v6.0)
> Passed!  - Failed:     0, Passed:  1162, Skipped:     2, Total:  1164, Duration: 9 m 10 s - Azure.DataApiBuilder.Service.Tests.dll (net6.0)

Reply:
Updated so that the objectConverter isn't used in the .net 8 test. I now don't need to ignore the test at all. Test passes for both .net6 and .net8.